arch/sim/src/sim/up_usestack.c

```diff
@@ -113,4 +114,11 @@ int up_use_stack(FAR struct tcb_s *tcb, FAR void *stack, size_t stack_size)
 #endif
 
   return OK;
+#else
+  /* Ignore the stack argument and allocate a new stack instead
+   * since it's impossible to extend a preallocated memory in place.
+   */
+
+  return up_create_stack(tcb, stack_size, tcb->flags & TCB_FLAG_TTYPE_MASK);
```

Comment:
`up_tls_size() + stack_size`?

Reply:
up_create_stack already consider this case.

Reply:
it's usually the responsibility of the caller of up_create_stack to adjust for up_tls_size, isn't it?

https://github.com/apache/incubator-nuttx/blob/a1af6ece3f6c06679c5be3ac3e9e956dc5d91a82/sched/task/task_init.c#L114-L127

Reply:
Yes, you are right. Done.